add npm weekly subscription opt-in to sign up

linclark opened the pull request:
Will fix #555 when it's done.

I figured I'd take a shot :) This is a first pass at adding mailchimp to the sign up flow. The `xxx` in there is a placeholder for the auth token. Where do we put that kind of config?

Also, it doesn't do anything on success or error. I don't think we want to display anything to the user, but should we log errors?

zeke commented on services/user/methods/signupUser.js:
```diff
@@ -1,9 +1,15 @@
 
 module.exports = function signupUser (acct, next) {
+  if (acct.npmweekly === "on") {
+    var mc = new mailchimp.Mailchimp('xxx');
```
If we're adding a secret here, it'll need to go into neww-config too, in the `env` stanza:

- https://github.com/npm/newww-config/blob/master/production/service.json
- https://github.com/npm/newww-config/blob/master/staging/service.json

zeke replied:
This `xxx` should be `process.env.MAILCHIMP_API` or similar.

zeke commented:
I wish we parsed locale strings as markdown. That way you could do "Sign me up for [npm Weekly](...) emails". But that's of course not a blocker.

zeke commented:
@linclark are you comfortable adding some tests for this? If not, I can help.

linclark commented:
Forgot to mention, this could possibly use some CSS adjustments.

![npm](https://cloud.githubusercontent.com/assets/105464/5829011/c3e5f5dc-a0d9-11e4-9071-c049d344f00b.png)

zeke commented:
Try putting a div around it with `.radio-group` class:

![screen shot 2015-01-20 at 4 29 13 pm](https://cloud.githubusercontent.com/assets/2289/5829074/8fd82afc-a0c1-11e4-829a-1bf5e494439f.png)

And feel free to rename radio-group to something more general :)

zeke commented:
PS that screenshot above is from https://www.npmjs.com/contact, which incidentally has some oddly formatted humans right now. So many css regressions. So little time!

linclark commented:
![npm](https://cloud.githubusercontent.com/assets/105464/5846989/5eb8828e-a196-11e4-9845-665fb1dc0b0a.png)

Thanks for the tip, looks good to me :)

linclark commented:
Ok, I ended up going with Sinon's spy. I modified a suggestion from CJ and basically added something analogous to setter injection. I also tried writing the test with nock and left the commit there in case folks want to look.

I wasn't able to add the test as part of `signup.js` since that mocks out the actual `signupUser` method, which is what I wanted to test. Instead, I added the tests to the user service tests. 

It currently only tests that the call to Mailchimp is made based on the checkbox, but it would be possible to test whether the users are created as well.

zeke commented:
Tests passing; works in the browser; looks good to me!

One suggestion: add `MAILCHIMP_KEY=12345` to `.env.example`

:shipit:
